### PR TITLE
Tested cancel job code. Added test code. TODO: Add scenario record

### PR DESCRIPTION
**Pull request by AditTulasi**

### CODE REVIEW

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/Jobs/StopAzureRmRecoveryServicesJob.cs [hunk 1]
```diff
@@ -23,17 +23,17 @@
...
 namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets
 {
-    [Cmdlet("Stop", "AzureRmBackupJob", DefaultParameterSetName = JobFilterSet), OutputType(typeof(AzureRmRecoveryServicesJobBase))]
+    [Cmdlet("Stop", "AzureRmRecoveryServicesJob", DefaultParameterSetName = JobFilterSet), OutputType(typeof(AzureRmRecoveryServicesJobBase))]
     public class StopAzureRmRecoveryServicesJob : RecoveryServicesBackupCmdletBase
     {
         protected const string IdFilterSet = "IdFilterSet";
         protected const string JobFilterSet = "JobFilterSet";
 
-        [Parameter(Mandatory = true, HelpMessage = ParamHelpMsg.Job.StopJobJobFilter, ParameterSetName = IdFilterSet)]
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/Jobs/StopAzureRmRecoveryServicesJob.cs]: Add position to the params

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Test/ScenarioTests/IaasVm/JobTests.ps1 [hunk 2]
```diff
@@ -123,23 +123,36 @@ function Test-GetJobDetails
...
 		Assert-AreNotEqual $waitedJob.Status "Cancelling"
 	}
+}
+
+function Test-CancelJobScenario
+{
+	$startTime = Get-Date -Date "2016-04-04 21:00:00"
+	$endTime = Get-Date -Date "2016-04-05 13:40:00"
+
+	$runningJobs = Get-AzureRmRecoveryServicesJob -From $startTime -To $endTime -Status "InProgress"
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Test/ScenarioTests/IaasVm/JobTests.ps1]: we should pass operation filter as backup or restore here

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/Jobs/StopAzureRmRecoveryServicesJob.cs [hunk 3]
```diff
@@ -48,6 +48,8 @@ public override void ExecuteCmdlet()
                     JobId = Job.InstanceId;
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/Jobs/StopAzureRmRecoveryServicesJob.cs]: Check with PM if we require confirmation before cancelling the job.

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/Jobs/StopAzureRmRecoveryServicesJob.cs (hunk 1 above)
Comment (AditTulasi) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/Jobs/StopAzureRmRecoveryServicesJob.cs]: Planning to do it for all cmdlets during stabilization.

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Test/ScenarioTests/IaasVm/JobTests.ps1 (hunk 2 above)
Comment (AditTulasi) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Test/ScenarioTests/IaasVm/JobTests.ps1]: Sure. This code has to be tested and modified accordingly.

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/Jobs/StopAzureRmRecoveryServicesJob.cs (hunk 3 above)
Comment (AditTulasi) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/Jobs/StopAzureRmRecoveryServicesJob.cs]: Sure thing.

**Pull request merged**